registry/registry.go

```diff
@@ -147,6 +147,7 @@ func (registry *Registry) ListenAndServe() error {
 					return err
 				}
 			}
+			m.SetHosts(config.HTTP.TLS.LetsEncrypt.Hosts)
```

Comment:
Why not only call this if `len(config.HTTP.TLS.LetsEncrypt.Hosts) > 0` and avoid the vendor change?

Reply:
Because then you can never unset the host list unless you manually delete the cachefile, as stated above:

> Technically it would be possible to rewrite the code to not depend on the vendor fix, but then users would have to delete the letsencrypt cachefile after removing the hosts setting from their registry config, or it would be restored from the stale entry in the cachefile.